sdks/python/apache_beam/runners/worker/data_plane.py

```diff
@@ -295,28 +295,48 @@ class GrpcClientDataChannelFactory(DataChannelFactory):
   Caches the created channels by ``data descriptor url``.
   """
 
-  def __init__(self):
+  def __init__(self, credentials=None):
     self._data_channel_cache = {}
     self._lock = threading.Lock()
+    self._credentials = None
+    if credentials is not None:
+      logging.info('Creating secure channel creds.')
+      self._credentials = credentials
 
   def create_data_channel(self, remote_grpc_port):
     url = remote_grpc_port.api_service_descriptor.url
     if url not in self._data_channel_cache:
       with self._lock:
         if url not in self._data_channel_cache:
           logging.info('Creating channel for %s', url)
-          grpc_channel = grpc.insecure_channel(
-              url,
-              # Options to have no limits (-1) on the size of the messages
-              # received or sent over the data plane. The actual buffer size is
-              # controlled in a layer above.
-              options=[("grpc.max_receive_message_length", -1),
-                       ("grpc.max_send_message_length", -1)])
-          # Add workerId to the grpc channel
-          grpc_channel = grpc.intercept_channel(grpc_channel,
-                                                WorkerIdInterceptor())
-          self._data_channel_cache[url] = GrpcClientDataChannel(
-              beam_fn_api_pb2_grpc.BeamFnDataStub(grpc_channel))
+          if self._credentials is not None:
+            grpc_channel = grpc.secure_channel(
+                url,
+                self._credentials,
+                # Options to have no limits (-1) on the size of the messages
+                # received or sent over the data plane. The actual buffer size
+                # is controlled in a layer above.
+                options=[("grpc.max_receive_message_length", -1),
```

Comment:
Create options external to the if branching.

Reply:
done.